packages/graphback/src/database/migrations/DataResourcesManager.ts

```diff
@@ -0,0 +1,379 @@
+// tslint:disable: await-promise
+import { Change, ChangeType } from '@graphql-inspector/core';
+import { unlinkSync } from 'fs';
+import { GlobSync } from 'glob';
+import * as Knex from 'knex';
+import { InputModelFieldContext, InputModelTypeContext } from '../../input/ContextTypes';
+import { logger } from '../../utils/logger';
+import { DatabaseContextProvider } from './DatabaseContextProvider';
+/**
+ * Represents update for data type
+ */
+export interface IDataLayerUpdate {
+  oldType: InputModelTypeContext,
+  newType: InputModelTypeContext
+}
+
+/**
+ * Interface for creating atabase resources .
+ * For example for relational databases implementors will execute DDL queries into database
+ */
+export interface IDataLayerResourcesManager {
```

Comment:
Actually this interface is very strongly implying some specific details - as it assumes that we always create relationships (for mongo it might not be case). Let's not change that until we figure out metadata and database part. Just flagging